src/main/java/Duke.java

```diff
@@ -1,10 +1,176 @@
+import java.util.Scanner;
+import java.util.*;
+import java.io.*;
+
 public class Duke {
-    public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+
+    private Storage storage;
+    private ArrayList<Task> allTasks = new ArrayList<Task>();
+    private Ui ui;
+
+    /**
+     *
+     * @param filePath
+     */
+    public Duke(String filePath) {
+        ui = new Ui();
+        storage = new Storage(filePath);
+
+        try {
+            File dataDir = new File("./data");
+            File dataText = new File("./data/duke.txt");
+            if (!dataDir.exists()){
+                dataDir.mkdirs();
+            }
+            if (!dataText.exists()){
+                dataText.createNewFile();
+            }
+            allTasks = new ArrayList<Task>(storage.load());
+        } catch (IOException e) {
+            ui.showLoadingError();
+            allTasks = new ArrayList<Task>();
+
+        }
+    }
+
+    public void run() throws IOException {
+        boolean notQuit = true;
+        int index;
+        String description, detail, by, at;
+        Task t;
+
+        ui.greeting();
+
+        while (notQuit) {
+            String command = ui.readCommand().replaceAll("( )+", " ");
+            switch (command.split(" ")[0]) {
+                case "bye":
+                    ui.sayGoobye();
+                    storage.writeToDukeFile(allTasks);
+                    notQuit = false;
+                    break;
+
+                case "mark":
+                    index = Integer.parseInt(command.split(" ")[1].trim()) - 1;
+                    t = allTasks.get(index);
+                    ui.markAndDisplayTask(t);
+                    break;
+
+                case "unmark":
+                    index = Integer.parseInt(command.split(" ")[1].trim()) - 1;
+                    t = allTasks.get(index);
+                    ui.unmarkAndDisplayTask(t);
+                    break;
+
+                case "list":
+                    ui.displayListWithStatus(allTasks, allTasks.size());
+                    break;
+
+                case "find":
+                    ArrayList<Task> foundTasks = new ArrayList<Task>();
+                    String keyword = command.split(" ")[1].trim();
+                    for (int i = 0; i < allTasks.size(); i++){
+                        Task task = allTasks.get(i);
+                        if(task.description.contains(keyword)){
+                            foundTasks.add(task);
+                        }
+                    }
+                    ui.displayFoundTasks(foundTasks);
+                    break;
+
+                case "delete":
+                    index = Integer.parseInt(command.split(" ")[1].trim()) - 1;
```

Comment:
Might need to catch exception if variable is not an int if not your program will stop running (as the exception not in a try catch block). Also try to create more variables so that your code looks more readable. command.split(" ")[1] can be a variable at the top of the switch case statements.